cmd/infra/azure/create.go

```diff
@@ -220,6 +220,38 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 		l.Info("Successfully created vnet", "ID", result.VNetID)
 	}
 
+	if o.ManagedIdentitiesFile != "" {
+		managedIdentitiesRaw, err := os.ReadFile(o.ManagedIdentitiesFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read --managed-identities-file %s: %w", o.ManagedIdentitiesFile, err)
+		}
+		if err := yaml.Unmarshal(managedIdentitiesRaw, &result.ControlPlaneMIs.ControlPlane); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal --managed-identities-file: %w", err)
+		}
+
+		components := map[string]string{
+			cpo:           result.ControlPlaneMIs.ControlPlane.ControlPlaneOperator.ClientID,
+			ciro:          result.ControlPlaneMIs.ControlPlane.ImageRegistry.ClientID,
+			nodePoolMgmt:  result.ControlPlaneMIs.ControlPlane.NodePoolManagement.ClientID,
+			cloudProvider: result.ControlPlaneMIs.ControlPlane.CloudProvider.ClientID,
+			azureFile:     result.ControlPlaneMIs.ControlPlane.File.ClientID,
+			azureDisk:     result.ControlPlaneMIs.ControlPlane.Disk.ClientID,
+			ingress:       result.ControlPlaneMIs.ControlPlane.Ingress.ClientID,
+			cncc:          result.ControlPlaneMIs.ControlPlane.Network.ClientID,
+		}
+
+		for component, clientID := range components {
+			objectID, err := findObjectId(clientID)
+			if err != nil {
+				return nil, err
+			}
+			err = assignContributorRole(subscriptionID, resourceGroupName, nsgResourceGroupName, vnetResourceGroupName, component, objectID)
```

Comment:
Why do we need to do this?

Reply:
we need to give the serviceprincipals that are used to authenticate different components this role because we don't know the RG names before creating the  hosted clusters in the  e2e so we need this part to automatically give the SP permission for that HCs RGs.